### PR TITLE
Completed basic functionality

### PR DESCRIPTION
**Pull request by Dmigdol**

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -18,3 +18,16 @@ app.use('/api', router);
 
 app.use(express.static(path.join(__dirname, '../client/dist')));
 module.exports = app;
+<<<<<<< Updated upstream
```
Comment (cesargua) [inline server/index.js]: get rid of this part please, this will break the code

File: server/index.js [hunk 2]
```diff
@@ -18,3 +18,16 @@ app.use('/api', router);
...
+const PORT = process.env.PORT || 3000;
+// const PORT = process.env.PORT || 3000;
+
+app.listen(PORT);
+console.log(`Server listening at http://localhost:${PORT}`);
+// app.listen(PORT);
+// console.log(`Server listening at http://localhost:${PORT}`);
+
+module.exports = app;
+>>>>>>> Stashed changes
```
Comment (cesargua) [inline server/index.js]: And here

**Review by cesargua — CHANGES_REQUESTED** [review]
Take another look at index.js

**Review by cesargua — CHANGES_REQUESTED** [review]
Take another look at server/index.js

**Review by cesargua — APPROVED** [review]

**Review by Chugale — APPROVED** [review]
should try and avoid using px if possible

File: client/src/components/questions_and_answers/Answers.jsx [hunk 3]
```diff
@@ -1,19 +1,47 @@
...
+  // Finishing component thought; small detail is show answers should not render if 2 or less answers
+
+  const handleAUpvote = (answerId) => {
+    axios.put(`/api/q_a/answer/${answerId}/upvote`, {
+      answer_id: answerId,
+    })
+      .then(() => {
+        console.log('ANSWER ', answerId, 'HAS BEEN UPOVOTED');
+      })
+      .catch((err) => {
```
Comment (cesargua) [inline client/src/components/questions_and_answers/Answers.jsx:21]: Good job on error catching. Very container safe!

File: client/src/components/questions_and_answers/Modal.jsx [hunk 4]
```diff
@@ -3,12 +3,21 @@ import axios from 'axios';
...
+function Modal({ closeModal, modalType, product_id }) {
+
+  console.log(product_id);
 
   const [username, setUsername] = useState('');
   const [email, setEmail] = useState('');
   const [body, setBody] = useState('');
 
+
+  const resetStates = () => {
```
Comment (cesargua) [inline client/src/components/questions_and_answers/Modal.jsx:15]: I like how you used a helper function on resetting states, very smart idea.

File: client/src/components/questions_and_answers/Search.jsx [hunk 5]
```diff
@@ -1,21 +1,34 @@
...
+  const [searchInput, setSearchInput] = useState('');
+  const [filteredList, setFilteredList] = useState('');
+
+  const handleChange = () => {
     // As soon as entry hits 3 chars, Begin filtering the List of questions
     // To include on relevant questions
-    if (text.length > 3) {
-      questions
+    if (searchInput.length > 2) {
+      setFilter(questions.filter((question) => question.question_body.match(searchInput)));
```
Comment (cesargua) [inline client/src/components/questions_and_answers/Search.jsx:15]: Although this was a good one liner, you could split this one up to lessen complexity and clarity.

**Pull request merged**